### PR TITLE
Final versionless fixes

### PR DESCRIPTION
**Pull request by rsherget**
OpenLiberty Pull Requester,

ATTENTION, READ THIS: Updated 4/11/2018 - Read and understand this completely,
then delete this entire template. If a reviewer or merger sees this template,
they should fail the review or merge.

If this code change is fixing a user-visible bug in previously released code, it MUST
have an associated issue mentioned in the PR text or description. That Issue also
MUST be labelled “release bug”

This directs automation to scrape this fix for inclusion in the next release's
list of bugs fixed.

If this code is NOT for fixing a released bug, for example new function, fixing
a bug in unreleased function, or other improvements that do not affect the
user-space, no label is needed. An issue could or could not be used based
on the developer’s discretion, but do still delete this text block.

For full details, please see this wiki page:

https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions

### CODE REVIEW

**Comment by rsherget** [comment]
#build

**Comment by rsherget** [comment]
build link https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_y4fE0BLDEe-GLt3IcvbSMw

**Comment by rsherget** [comment]
#build

File: dev/build.featureStart.base/src/com/ibm/ws/test/featurestart/features/FeatureFilter.java [hunk 1]
```diff
@@ -67,9 +67,8 @@ public static String skipFeature(String shortName) {
             // location variable.
             return "Required variable 'wmqJmsClient.rar.location' is not set";
 
-        } else if (shortName.equalsIgnoreCase("mpHealth") ||
-                   shortName.equalsIgnoreCase("mpMetrics")){
-            // Versionless feature cannot run by itself
+        } else if (shortName.indexOf("-") == -1 ) {
```
Comment (cbridgha) [inline dev/build.featureStart.base/src/com/ibm/ws/test/featurestart/features/FeatureFilter.java:70]: We should add a helperMethod isVersionless(String) to make it obvious what this is doing...

File: dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/FeatureManager.java [hunk 2]
```diff
@@ -66,6 +67,7 @@
 import org.osgi.framework.wiring.BundleRevision;
 import org.osgi.framework.wiring.FrameworkWiring;
 import org.osgi.resource.Requirement;
+//import org.osgi.service.cm.ConfigurationException;
 import org.osgi.service.cm.ConfigurationException;
```
Comment (cbridgha) [inline dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/FeatureManager.java:71]: should delete

**Review by cbridgha — APPROVED** [review]
Just to make sure...  there are a dozen new private ga features like "io.openliberty.internal.mpVersion-1.0".  this is intended?   Or should they also be marked no-ship?

**Comment by rsherget** [comment]
#libby

**Comment by rsherget** [comment]
> Just to make sure... there are a dozen new private ga features like "io.openliberty.internal.mpVersion-1.0". this is intended? Or should they also be marked no-ship?

This is intended. These features are brought in by mpCompatible features, which are already marked ga, so mpVersion features also must be ga. There is no difference in resolution of existing features when these features are introduced.

**Comment by rsherget** [comment]
Only one build fat test error and it is unrelated to these changes, merging

**Pull request merged**